infra/discovery/src/lib/search.js

```diff
@@ -298,13 +336,53 @@ class Listing {
       numberOfItems = 1000
     }
 
+    // sorting
+    const getSortQuery = () => {
+      // sort  returns as [Object null prototype] {target:, order: } not sure why
+      // the following code converts it to a regular object
+      const resolvedSort = JSON.parse(JSON.stringify(sort))[0]
+      const { target, order } = resolvedSort
+      // if target and order are set, return a sort
+      // otherwise return empty sort to skip
+      if (target && order) {
+        switch (target) {
+          case 'price.amount':
+            return {
+              _script: {
+                type: 'number',
+                script: {
+                  lang: 'painless',
+                  source: `Float.parseFloat(params._source.${target}) * Float.parseFloat(params.exchangeRates[params._source.price.currency.id])`,
+                  params: {
+                    exchangeRates: exchangeRates
+                  }
+                },
+                order: order
+              }
+            }
+          default:
+            // this default is for non script based sorting, should satisfy most cases
+            return [
+              {
+                [target]: {
```

Comment:
I wonder what kind of evil fun we can have here, with the ability to sort on any user chosen field in our data. Extract moderation scores?

Reply:
There's a lot of possibilities, we could get quite deep here. There is also multi value based sorting, such as an array of different fields, or [applying modes (min,max,sum,average,median)](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html#_sort_mode_option). This code could be easily expanded upon.

Reply:
I'll add a whitelist and error if a value is entered outside of that.